controller_manager/scripts/controller_group

```diff
@@ -0,0 +1,96 @@
+from controller_manager_msgs.srv import ListControllers, ListControllersRequest
+
+
+class Groups(object):
+    @staticmethod
+    def list(args):
+        groups = rospy.get_param('/controller_groups', {})
+        print('Number of groups: {}'.format(len(groups)))
+        for group, controllers in groups.iteritems():
+            print('  {}'.format(group))
+            for controller in controllers:
+                print('      {}'.format(controller))
+
+    @staticmethod
+    def spawn(args):
+        desired_controllers, _ = Groups._read(args.group)
+        if not desired_controllers:
+            return
+        for controller in desired_controllers:
+            controller_manager_interface.load_controller(controller)
+        for controller in desired_controllers:
+            controller_manager_interface.start_controller(controller)
+
+    @staticmethod
+    def switch(args):
+        desired_controllers, other_controllers = Groups._read(args.group)
+        if not desired_controllers:
+            return
+        loaded_controllers = Groups._get_loaded_controllers()
+        # Stop running controllers that we no longer need.
+        for controller in other_controllers:
+            if loaded_controllers.get(controller, False):
+                controller_manager_interface.stop_controller(controller)
+
+        # Load and start every desired controller if needed.
+        for controller in desired_controllers:
+            if controller not in loaded_controllers:
+                controller_manager_interface.load_controller(controller)
+            if not loaded_controllers.get(controller, False):
+                controller_manager_interface.start_controller(controller)
+
+    @staticmethod
+    def _read(group):
+        groups = rospy.get_param('/controller_groups', {})
+        desired_controllers = set(groups.get(group, ()))
+        if not desired_controllers:
+            rospy.logerr('There is no controller defined in group {}'.format(group))
+            return None, None
+        all_controllers = set()
+        for controllers in groups.values():
+            # Convert list to set to remove duplicates.
+            all_controllers = all_controllers | set(controllers)
+        return desired_controllers, all_controllers - desired_controllers
+
+    @staticmethod
+    def _get_loaded_controllers():
+        rospy.wait_for_service('controller_manager/list_controllers')
+        s = rospy.ServiceProxy('controller_manager/list_controllers', ListControllers)
+        resp = s.call(ListControllersRequest())
+        controllers = {}
+        for controller in resp.controller:
+            controllers[controller.name] = controller.state == 'running'
+            if controller.state not in ('running', 'stopped'):
+                rospy.logerr('Unknown controller state {}'.format(controller.state))
+        return controllers
+
+
+def _main():
+    rospy.init_node('controller_group', anonymous=True)
```

Comment:
Initializing a node may actually not be necessary, since you're only calling services and not interacting with topics at all.

Reply:
Added for rospy.myargv() but seems not necessary.

Reply:
That's a good point— this definitely does still need to work when run from a launch file.

Reply:
do you mean init_node is still needed?

Reply:
I don't think it should be; `myargv` will have you covered— was more just acknowledging that that use case be considered.